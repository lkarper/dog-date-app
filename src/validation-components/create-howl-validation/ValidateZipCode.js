import React, { useEffect } from 'react';

const ValidateZipCode = (props) => {
    
    const { zipcode, zipcodeError, setZipcodeError } = props;

    useEffect(() => {
        
        if (!/(^[0-9]{5}$)|(^[0-9]{5}-[0-9]{4}$)/.test(zipcode)) {
            setZipcodeError(`ZIP code is required and must match format '12345' or '12345-6789'.`);
        } else {
            setZipcodeError('');
        }

    }, [zipcode, setZipcodeError]);

    if (zipcodeError) {
        return <p id='zipcode-validator'>{zipcodeError}</p>;
    }

    return <p id='zipcode-validator'>ZIP code set to {zipcode}.</p>;
}

export default ValidateZipCode;