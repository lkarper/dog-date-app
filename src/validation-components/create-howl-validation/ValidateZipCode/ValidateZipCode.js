import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateZipCode = (props) => {
    
    const { 
        zipcode, 
        zipcodeError, 
        setZipcodeError, 
        notRequired
    } = props;

    useEffect(() => {  
        if (notRequired && zipcode.length === 0) {
            setZipcodeError('');
        } else if (!/^[0-9]{5}$/.test(zipcode)) {
            if (notRequired) {
                setZipcodeError(`Zip code must be 5 digits in length if provided.`);
            } else {
                setZipcodeError(`ZIP code is required and must be 5 digits in length.`);
            } 
        } else {
            setZipcodeError('');
        }
    }, [zipcode, setZipcodeError, notRequired]);

    if (zipcodeError) {
        return (
            <p 
                className='ValidateZipCode__validator error'
                id='zipcode-validator'
            >
                {zipcodeError}
            </p>
        );
    }

    if (notRequired && !zipcodeError && !zipcode) {
        return (
            <p
                className='ValidateZipCode__validator' 
                id='zipcode-validator'
            >
                Zip code is optional, but must be 5 digits in length if provided.
            </p>
        );
    }

    return (
        <p 
            className='ValidateZipCode__validator valid'
            id='zipcode-validator'
        >
            ZIP code set to '{zipcode}'.
        </p>
    );
}

ValidateZipCode.defaultProps = {
    zipcode: '', 
    zipcodeError: `ZIP code is required and must be 5 digits in length.`, 
    setZipcodeError: () => {}, 
    notRequired: false, 
};

ValidateZipCode.propTypes = {
    zipcode: PropTypes.string.isRequired,
    zipcodeError: PropTypes.string.isRequired,
    setZipcodeError: PropTypes.func.isRequired,
    notRequired: PropTypes.bool.isRequired,
};

export default ValidateZipCode;
