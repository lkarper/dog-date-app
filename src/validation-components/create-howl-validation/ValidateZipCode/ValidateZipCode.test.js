import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateZipCode from './ValidateZipCode';

describe('ValidateZipCode component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateZipCode />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateZipCode />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
