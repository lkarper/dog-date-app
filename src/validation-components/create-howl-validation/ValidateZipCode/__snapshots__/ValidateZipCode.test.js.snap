// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateZipCode component renders the UI as expected 1`] = `
<ValidateZipCode
  notRequired={false}
  setZipcodeError={[Function]}
  zipcode=""
  zipcodeError="ZIP code is required and must be 5 digits in length."
>
  <p
    className="ValidateZipCode__validator error"
    id="zipcode-validator"
  >
    ZIP code is required and must be 5 digits in length.
  </p>
</ValidateZipCode>
`;
